src/plugins/spdk/spdk_progress_engine.cpp

```
@@ -0,0 +1,1195 @@
+            result.emplace(fn());
+        });
+        done.wait();
+        return std::move(*result);
+    }
+}
+
+void
+nixlSpdkProgressEngine::enqueueAsync(std::function<void()> fn) {
+    {
+        const std::scoped_lock lock(queueMutex_);
+        producerQueue_.push_back(std::move(fn));
```

Cppcheck (2.21.0)

[error] Non-local variable 'producerQueue_' will use object that points to local variable 'done'.

(danglingLifetime)